workshops/pyTelegramWorkshop/README.md

```diff
@@ -0,0 +1,378 @@
+
+When you are done with trying this out, proceed ahead.
+
+
+
+## DISECTING THE CODE
+
+Now that we have experienced how seamlessly the WEB API works (I hope you did not skip), let us trace down every line: 
+
+This is requests, an alternative to Python’s built-in URLLIB. This needs to be installed via *_pip_* (command in workshop requisites section) if you’re running this off your own PC. 
+
+Running this code on repl.it doesn’t require you to pip.
+
+```python
+import requests
+```
+
+Now we define a function, *_sendMsg()_*. To call this function, we will need to pass three parameters with it, your *_bot’s token_*, your *_chat ID_* and the text you need to send. As simple as that!
+
+_send_text_ concatenates your token, chat ID and message with the base URL of Telegram’s API. This includes the method *_sendMessage_*, which is used to… well, send messages.
+
+Now we define and call a variable, response, which allows us to use the requests module’s get function to access the URL we stored in the variable _send_text_, in the above step.
+
+Now we just return *_response.json_* in an effort to improve our code and unify things.
+
+And…. that’s it!
+
+```python
+def sendMsg(token, chatID, text):
+  #Making the URL we need by concatinating various parts
+  send_text = 'https://api.telegram.org/bot' + token + '/sendMessage?chat_id=' + chatID + '&parse_mode=Markdown&text=' + text
+  response = requests.get(send_text) #'.get' sends and returns the response (which is none here)
+  print("Sent! ") #confirmation of function call
+  return response.json() #should work just as well even without returning
+```
+
+What follows is just a bunch of input() statements to receive values from the user to pass-on to the function.
+
+```python
+#Token ID of the bot as recieved from BotFather
+print("You'll now be asked for your BOT ID. Enter it without quotes. ")
+
+token = input("Enter your bot token: ")
+#Get your chat ID by sending pinging this bot: https://t.me/get_id_bot
+chatID = input("Enter your (own) Telegram chat ID: ")
+```
+
+Then we create a text variable with out text, and simple pass it to the function!
+
+```python
+#Final function call
+sendMsg(token, chatID, text)
+```
+
+## RECEIVING MESSAGES
+
+Remember using *_sendMessage_* in the URL when sending a message via your bot?
+
+To receive messages, we use *_getUpdates_*. Telegram handles receiving messages brilliantly! Whenever you call *_getUpdates_* with your bot token, it returns a .JSON with all your queued messages.
+
+Why do I say “queued”? Because your messages don’t clear till you issue a “receipt” to Telegram of the same.
+
+Each message comes with an *_updateID_*, which you can return to Telegram to clear that from the queue. For example, if you have 10 messages, simply passing the *_updateID_* of the latest message clears all 10 messages.
+
+Yeah, this might sound complex, but this small demonstration on repl.it should make it seem all easy-peasy!
```

Comment:
```suggestion
Remember using *_sendMessage_* in the URL when sending a message via your bot?

To receive messages, we use *_getUpdates_*. Telegram handles receiving messages brilliantly! Whenever you call *_getUpdates_* with your bot token, it returns a .JSON with all your queued messages.

Why do I say “queued”? Because your messages don’t clear till you issue a “receipt” to Telegram of the same.

Each message comes with a *_updateID_*, which you can return to Telegram to clear that from the queue. For example, if you have 10 messages, simply passing the *_updateID_* of the latest message clears all 10 messages.

Yeah, this might sound complex, but this small demonstration on repl.it should make it seem all easy-peasy!
```